pkg/rbd/controllerserver.go

```diff
@@ -129,6 +129,36 @@ func (cs *ControllerServer) parseVolCreateRequest(ctx context.Context, req *csi.
 	return rbdVol, nil
 }
 
+func (cs *ControllerServer) getExistingVolume(ctx context.Context, req *csi.CreateVolumeRequest, rbdVol *rbdVolume) (*csi.CreateVolumeResponse, error) {
+	if rbdVol.Encrypted {
+		err := rbdVol.ensureEncryptionMetadataSet(rbdImageRequiresEncryption)
+		if err != nil {
+			klog.Errorf(util.Log(ctx, err.Error()))
+			return nil, err
```

Comment:
ensureEncryptedMetadataSet returns fmt.Errorf. need to return. GRPC errorcode and message to consumer

Reply:
Done with the next update.